src/libfetchers/git.cc

```diff
@@ -244,6 +244,12 @@ std::pair<StorePath, Input> fetchFromWorkdir(ref<Store> store, Input & input, co
         "lastModified",
         workdirInfo.hasHead ? std::stoull(runProgram("git", true, { "-C", actualPath, "--git-dir", gitDir, "log", "-1", "--format=%ct", "--no-show-signature", "HEAD" })) : 0);
 
+    // QUESTION: should this just return an empty hash if there are no commits?
+    if (workdirInfo.hasHead) {
+        input.attrs.insert_or_assign("dirtyRev", chomp(
+            runProgram("git", true, { "-C", actualPath, "--git-dir", gitDir, "describe", "--always", "--match=''", "--abbrev=0", "--dirty" })));
```

Comment:
This command is surprisingly slow in github:NixOS/nixpkgs, I'm really not sure why given the `--match=''` means it always ends up outputting the full sha, and it's not the `--dirty` flag that makes it slow (it's slow without that too).

I'm inclined to say this should instead run `git rev-parse --verify HEAD` for the sha, and then some other command to compute dirty status (perhaps `git diff --quiet HEAD`).

I'm also not sure how submodules affect this. AIUI current behavior is to ignore submodules unless the flake is specified with `?submodules=1`, but I don't know if that affects the current revision calculation or if that propagates to this command somehow.

Reply:
I've chosen to switch this to use `git rev-parse --verify HEAD` as suggested :+1: 

As this code only runs if the worktree is dirty so we don't need to check that it's dirty and we can manually append "-dirty" unconditionally